### PR TITLE
Automated tests (part 1) (connect #613)

### PR DESCRIPTION
**Pull request by MelEnt**
#### Before the PR (what is the issue or what needed to be done)
Automated UI tests using Espresso framework to ensure basic functionality

#### The solution
Test an instance of free text survey and make sure it's possible to install it and properly edit it

#### Reviewer Checklist
* [ ] Connect the issue
* [ ] Test plan
* [ ] Copyright header
* [ ] Code formatting
* [ ] Documentation

### CODE REVIEW

File: app/src/androidTest/java/org/akvo/flow/activity/FreeTextSurveyTest.java [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+
+    @Rule
+    public ActivityTestRule<SurveyActivity> rule = new ActivityTestRule<>(SurveyActivity.class);
+
+    private Context runtimeContext;
+    private SurveyInstaller installer;
+
+    @Before
+    public void init()
+    {
```
Comment (valllllll2000) [inline app/src/androidTest/java/org/akvo/flow/activity/FreeTextSurveyTest.java]: please fix the brackets formatting

File: app/src/androidTest/java/org/akvo/flow/activity/FreeTextSurveyTest.java [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+        return survey;
+    }
+
+    private Survey getSurvey(int resId) throws IOException {
+        InputStream input = runtimeContext.getResources().openRawResource(resId);
+        return installer.persistSurvey(FileUtil.readText(input));
+    }
+
+    private void openDrawer()
+    {
```
Comment (valllllll2000) [inline app/src/androidTest/java/org/akvo/flow/activity/FreeTextSurveyTest.java]: please fix the brackets formatting

File: app/src/androidTest/java/org/akvo/flow/testhelper/SurveyInstaller.java [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+import java.io.IOException;
+import java.io.Writer;
+
+public class SurveyInstaller {
+
+    private SurveyDbAdapter adapter;
+    private Context context;
+
+    public SurveyInstaller(Context context)
+    {
```
Comment (valllllll2000) [inline app/src/androidTest/java/org/akvo/flow/testhelper/SurveyInstaller.java]: brackets formatting

File: app/src/androidTest/java/org/akvo/flow/activity/FreeTextSurveyTest.java [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+    public void ensureCantSubmitEmptyFreeText() throws IOException {
+        fillFreeTextQuestion(freetextsurvey, "");
+        onView(allOf(withClassName(endsWith("Button")), withText(R.string.submitbutton))).check(matches(not(isEnabled())));
+    }
+
+    private Survey fillFreeTextQuestion(int surveyResId, String text) throws IOException {
+        Survey survey = getSurvey(surveyResId);
+
+        openDrawer();
+        //Check for specified survey name, make sure it matches what's displayed and click it
```
Comment (valllllll2000) [inline app/src/androidTest/java/org/akvo/flow/activity/FreeTextSurveyTest.java]: lines 83-92: I would say these comments are not necessary as it seems obvious what the code does. When it is not obvious what the code does it is better to refactor it and name the method in such a way that is self explanatory. See for example this blog https://blog.codinghorror.com/coding-without-comments/.

Comment (MelEnt) [inline app/src/androidTest/java/org/akvo/flow/activity/FreeTextSurveyTest.java]: Thanks for the article!

**Pull request merged**